src/Storage/ParsedMail.hs

```diff
@@ -7,18 +7,21 @@ import Codec.MIME.Parse (parseMIMEMessage)
 import Codec.MIME.Type
        (MIMEParam(..), MIMEValue(..), mime_val_headers)
 import Control.Exception (try)
-import Control.Lens.Getter (view)
-import qualified Data.Text           as T
+import qualified Data.Text as T
 import Data.Text.IO (readFile)
+import Data.Monoid ((<>))
 import Prelude hiding (readFile)
-import Types (ParsedMail(..), NotmuchMail, mailFilepath)
+import Storage.Notmuch (mailFilepath)
+import Types (ParsedMail(..), NotmuchMail)
 
-parseMail :: NotmuchMail -> IO (Either String ParsedMail)
-parseMail m = do
-    msg <- try (readFile $ view mailFilepath m) :: IO (Either IOError T.Text)
-    case msg of
-        Left e -> pure $ Left $ show e
-        Right contents -> pure $ Right $ MIMEMail (parseMIMEMessage contents)
+parseMail :: NotmuchMail
+          -> FilePath
+          -> IO (Either String ParsedMail)
+parseMail m dbpath =
+    mailFilepath m dbpath
+           >>= either fail (\x -> try (readFile x):: IO (Either IOError T.Text))
+           >>= either (\x -> pure $ Left (show x <> " Maybe run notmuch new again?")) (pure . Right)
+           >>= either (pure . Left . show) (pure . Right . MIMEMail <$> parseMIMEMessage)
```

Comment:
`fail` is bad bad bad.

Do this instead:
```haskell
parseMail m dbpath =                                                                                     
  mailFilepath m dbpath                                                                                  
  >>= fmap (bimap (\x -> show x <> " Maybe run notmuch new again?") (MIMEMail . parseMIMEMessage) . join)
      . traverse (fmap (first (show :: IOError -> String)) . try . T.readFile)                           
```

The whole thing will improve once I do the "abstract over error type" refactor, though.